do/zones.go

```diff
@@ -17,19 +17,47 @@ limitations under the License.
 package do
 
 import (
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/godo/context"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/kubernetes/pkg/cloudprovider"
 )
 
 type zones struct {
+	client *godo.Client
 	region string
 }
 
 func newZones(region string) cloudprovider.Zones {
 	return zones{region}
 }
 
-// GetZone returns a cloudprovider.Zone
-// GetZone will only fill the Region field of cloudprovider.Zone for DO
+// GetZone returns a cloudprovider.Zone. GetZone will only fill the Region
+// field of cloudprovider.Zone for DO.
 func (z zones) GetZone() (cloudprovider.Zone, error) {
 	return cloudprovider.Zone{Region: z.region}, nil
 }
+
+// GetZoneByProviderID returns the Zone containing the current zone and
+// locality region of the node specified by providerId. GetZoneByProviderID
+// will only fill the Region field of cloudprovider.Zone for DO.
+func (z zones) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
```

Comment:
This is actually going to depend on some of the changes in https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/36